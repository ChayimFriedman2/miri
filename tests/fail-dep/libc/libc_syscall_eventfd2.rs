//@only-target: linux

// This is a test for calling eventfd2 through a syscall.
// But we do not support this.
fn main() {
    let initval = 0 as libc::c_uint;
    let flags = (libc::EFD_CLOEXEC | libc::EFD_NONBLOCK) as libc::c_int;

    let result = unsafe {
        libc::syscall(libc::SYS_eventfd2, initval, flags) //~ERROR: unsupported operation
    };

    assert_eq!(result, 3); // The first FD provided would be 3.
}
