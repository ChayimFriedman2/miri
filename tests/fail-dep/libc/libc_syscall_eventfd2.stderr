error: unsupported operation: can't execute syscall with ID 290
  --> tests/fail-dep/libc/libc_syscall_eventfd2.rs:LL:CC
   |
LL |         libc::syscall(libc::SYS_eventfd2, initval, flags)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't execute syscall with ID 290
   |
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
   = note: BACKTRACE:
   = note: inside `main` at tests/fail-dep/libc/libc_syscall_eventfd2.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error

